products/compute/api.yaml

```diff
@@ -935,6 +935,266 @@ objects:
+        min_version: beta
+        item_type: Api::Type::String
+        description: |
+          Headers that the HTTP/S load balancer should add to proxied
+          requests.
+      - !ruby/object:Api::Type::Fingerprint
+        name: 'fingerprint'
+        output: true
+        description: |
+          Fingerprint of this resource. A hash of the contents stored in this
+          object. This field is used in optimistic locking.
+      - !ruby/object:Api::Type::String
+        name: 'description'
+        description: |
+          An optional description of this resource.
+      - !ruby/object:Api::Type::Array
+        name: 'healthChecks'
+        item_type: Api::Type::String
+        required: true
+        min_size: 1
+        max_size: 1
+        description: |
+           The list of URLs to the HttpHealthCheck or HttpsHealthCheck resource
+           for health checking this BackendService. Currently at most one health
+           check can be specified, and a health check is required.
+
+           For internal load balancing, a URL to a HealthCheck resource must be
+           specified instead.
+      - !ruby/object:Api::Type::Integer
+        name: 'id'
+        description: 'The unique identifier for the resource.'
+        output: true
+      - !ruby/object:Api::Type::NestedObject
+        name: 'iap'
+        description: Settings for enabling Cloud Identity Aware Proxy
+        properties:
+          - !ruby/object:Api::Type::Boolean
+            name: 'enabled'
+            description: Enables IAP.
+          - !ruby/object:Api::Type::String
+            name: 'oauth2ClientId'
+            required: true
+            description: |
+              OAuth2 Client ID for IAP
+          - !ruby/object:Api::Type::String
+            name: 'oauth2ClientSecret'
+            required: true
+            description: |
+              OAuth2 Client Secret for IAP
+          - !ruby/object:Api::Type::String
+            name: 'oauth2ClientSecretSha256'
+            output: true
+            description: |
+              OAuth2 Client Secret SHA-256 for IAP
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        required: true
+        input: true
+        description: |
+          Name of the resource. Provided by the client when the resource is
+          created. The name must be 1-63 characters long, and comply with
+          RFC1035. Specifically, the name must be 1-63 characters long and match
+          the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?` which means the
+          first character must be a lowercase letter, and all following
+          characters must be a dash, lowercase letter, or digit, except the last
+          character, which cannot be a dash.
+      - !ruby/object:Api::Type::Enum
+        name: 'protocol'
+        description: |
+          The protocol this BackendService uses to communicate with backends.
+
+          The possible values are TCP and UDP, and the default is TCP.
```

Comment:
Do we prefer to allow the API to default this or default via `default_value`?

Reply:
We should set a default_value and then not use it in Terraform I think- adding one probably counts as a breaking change.

Reply:
Is it possible to override a default_value to remove it?

Reply:
No idea! If it's impossible, I'd comment out the `default_value` in the api.yaml and note that downstreams need to implement it as an override